src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -146,6 +146,14 @@ public SAMSequenceDictionary makeSequenceDictionary(final File referenceFile) {
         if (URI == null) {
             URI = "file:" + REFERENCE.getAbsolutePath();
         }
+        if (OUTPUT == null) {
+            // determine the name for the dict file in the same way as CachingIndexedFastaSequenceFile.checkAndCreate
+            final String name = REFERENCE.getName();
+            final String fastaExt = ReferenceSequenceFileFactory.FASTA_EXTENSIONS.stream()
+                    .filter(name::endsWith).findFirst().orElseGet(() -> "");
+            OUTPUT = new File(REFERENCE.getParentFile(),
+                    REFERENCE.getName().replace(fastaExt, IOUtil.DICT_FILE_EXTENSION));
```

Comment:
Also, in this case, please emit a info or warning message informing the user of the choice of output name.